File: services/auth/src/models/response.rs
use serde_json::{json, Value};

/// Standard API response structure for success cases
pub struct ApiResponse;

impl ApiResponse {
    /// Create a success response with data
    pub fn success(data: Value) -> Value {
        json!({
            "success": true,
            "data": data
        })
    }

    /// Create an error response with code and message
    pub fn error(code: &str, message: &str) -> Value {
        json!({
            "success": false,
            "error": {
                "code": code,
                "message": message
            }
        })
    }
}

File: services/notes/src/utils/response.ts
import type { TApiError } from "../types/proto_generated/_common/interface.types";

/**
 * Standard API response structure
 */
export class ApiResponse {
  /**
   * Create a success response with data
   */
  static success<T>(data: T) {
    return {
      success: true,
      data,
    };
  }

  /**
   * Create an error response with code and message
   */
  static error(code: string, message: string) {
    return {
      success: false,
      error: {
        code,
        message,
      } as TApiError,
    };
  }

  /**
   * Create a delete success response (no data)
   */
  static deleteSuccess() {
    return {
      success: true,
    };
  }
}
